rewrite/rewrite/markers.py

```diff
@@ -30,6 +30,8 @@ def __hash__(self) -> int:
         return hash(self.id)
 
 
+M = TypeVar('M', bound=Marker)
```

Comment:
Good call. Can you also use it in the signatures of `find_first` and `find_all`?

Reply:
Done